earth_enterprise/src/fusion/autoingest/plugins.src/RasterProject.src

```diff
@@ -575,6 +584,24 @@ void RasterProjectAssetImplD::HandleModifyRequest(
     project->config.is_timemachine_project_ = is_timemachine_project;
     notify(NFY_DEBUG, "TimeMachine: %s for %s\n", req.assetname.c_str(),
            is_timemachine_project ? "on" : "off");
+
+    // By default keep is_overlay_terrain_project_ the same unless directed
+    // otherwise.
+    bool is_overlay_terrain_project = project->config.is_overlay_terrain_project_;
+    if (req.enable_terrain_overlay){
+      is_overlay_terrain_project = true;
+    }
+    else if (req.disable_terrain_overlay){
+      is_overlay_terrain_project = false;
+    }
+    if (is_overlay_terrain_project) {
+      // Set start level and/or resources min level if specified
+      if (req.overlay_terrain_start_level != 0)
```

Comment:
we probably need to do a little more validation for start level and resource level:
- start level can't be negative and may exceed 24 or so
- same for resource level
- any relationship between start level and resource level ? if any does it need to be enforced ?

Reply:
I now see that there is validation done in ```options.opt("start_level", start_level, &khGetopt::IsEvenNumberInRange<uint, 4, 24>);``` Is that only place where arguments validation is needed ?

Reply:
I think that is sufficient. That guarantees that the user can't pass in anything outside that range. Additionally, if the user doesn't specify a value, it will have the default value that was assigned earlier in the function. Also, in the `RasterProjectAssetImplD::HandleXXXXRequest` methods, the values for the different options are only used if the user specified them.